setup.py

```diff
@@ -2201,15 +2201,21 @@ def detect_ctypes(self):
                 ffi_inc = None
                 print('Header file {} does not exist'.format(ffi_h))
         ffi_lib = None
+        ffi_libdir = [sysconfig.get_config_var("LIBFFI_LIBDIR")]
+        ffi_libdir = [d for d in ffi_libdir if os.path.isdir(d)]
         if ffi_inc is not None:
             for lib_name in ('ffi', 'ffi_pic'):
-                if (self.compiler.find_library_file(self.lib_dirs, lib_name)):
+                fullpath = self.compiler.find_library_file(self.lib_dirs + ffi_libdir, lib_name)
```

Comment:
Why use  setup.py heuristics to find libname when you could just let pkgconfig gives the right hint for compilation host/target with `pkg-config libffi --libs-only-l`  like you did with LIBFFI_LIBDIR and LIBFFI_INCLUDEDIR ?

Reply:
That is a reasonable question.

The true answer is that while making my changes I tried to keep them as small as practical.

However when I came here to respond, I notice that this library search checks for libffi and libffi_pic, whether or not pkg-config is available on the compilation host.

If we simply create a new configure variable with say:
```
LIBFFI_LIBNAME="`"$PKG_CONFIG" libffi --libs-only-l 2> /dev/null | sed -e 's/^-l//;s/ *$//'`"
```
then we will lose the ability to find `libffi_pic.{so,a}` if there's no pkg-config.

We could set a default like `LIBFFI_LIBNAME="ffi ffi_pic"` and then have setup.py split it's value on space before looping much as now...

My feeling is to stick as we are, but happy to defer to core team

Reply:
> whether or not pkg-config is available on the compilation host.

maybe if pkg-config is here then use it fully  else use old way ( and good luck for cross compilation setup.py will pick anything in /usr instead of sysroot in most cases ).